temporalio/worker/_workflow.py

```diff
@@ -250,9 +300,32 @@ async def _handle_activation(
                         activate_task, self._deadlock_timeout_seconds
                     )
                 except asyncio.TimeoutError:
-                    raise RuntimeError(
-                        f"[TMPRL1101] Potential deadlock detected, workflow didn't yield within {self._deadlock_timeout_seconds} second(s)"
-                    )
+                    # extract the stack of the workflow if possible
+
+                    # cases:
+                    # 1. get_worker_ident() is not implemented for the worker
+                    # 2. get_worker_ident() returns None
+                    # 3. tid is not present in sys._current_frames()
+                    # 4. error occurs during traceback construction
+                    # 5. normal
+
+                    msg = f"[TMPRL1101] Potential deadlock detected, workflow didn't yield within {self._deadlock_timeout_seconds} second(s)"
+                    try:
+                        tid = workflow.get_worker_ident()
+                    except NotImplementedError:
+                        # Default error if the function is not implemented for the workers
+                        raise RuntimeError(msg) from None
+
+                    tb, warn = get_traceback_for_thread(tid)
+
+                    if not tb:
+                        msg += "\n\tWorker was unable to retrieve workflow thread info. No frames from workflow available."
+                        raise RuntimeError(msg) from None
+                    else:
+                        if tb and warn:
+                            msg += "\n\System encountered errors while constructing traceback, minimal or partial traceback will be presented."
```

Comment:
Usually you want to avoid putting newlines in error messages. A period to separate the sentences should be fine.